cranelift/codegen/src/machinst/helpers.rs

```diff
@@ -0,0 +1,35 @@
+//! Miscellaneous helpers for machine backends.
+
+use crate::ir::types::*;
+use crate::ir::Type;
+
+/// Returns the size (in bits) of a given type.
+pub fn ty_bits(ty: Type) -> usize {
+    match ty {
+        B1 => 1,
+        B8 | I8 => 8,
+        B16 | I16 => 16,
+        B32 | I32 | F32 | R32 => 32,
+        B64 | I64 | F64 | R64 => 64,
+        B128 | I128 => 128,
+        IFLAGS | FFLAGS => 32,
+        B8X8 | I8X8 | B16X4 | I16X4 | B32X2 | I32X2 => 64,
+        B8X16 | I8X16 | B16X8 | I16X8 | B32X4 | I32X4 | B64X2 | I64X2 => 128,
+        F32X4 | F64X2 => 128,
+        _ => panic!("ty_bits() on unknown type: {:?}", ty),
+    }
```

Comment:
```suggestion
    usize::from(ty.bits())
```

Reply:
We're sure that `usize::from(ty.bits())` actucally produces the same numbers, yes?

Reply:
https://github.com/bytecodealliance/wasmtime/blob/a577667f9fcc29528532f11a17e5e747cba7b13e/cranelift/codegen/src/ir/types.rs#L246-L248

https://github.com/bytecodealliance/wasmtime/blob/a577667f9fcc29528532f11a17e5e747cba7b13e/cranelift/codegen/src/ir/types.rs#L69-L80

For `IFLAGS` and `FFLAGS` it always returns 0. Other than that everything produces the same value.